packages/flame_3d/lib/src/components/light_component.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flame_3d/camera.dart';
+import 'package:flame_3d/components.dart';
+import 'package:flame_3d/game.dart';
+import 'package:flame_3d/resources.dart';
+
+/// A [Component3D] that represents a light source in the 3D world.
+class LightComponent extends Component3D {
+  LightComponent({
+    required this.source,
+    super.position,
+  });
+
+  LightComponent.spot({
+    Vector3? position,
+  }) : this(
+          source: SpotLight(),
+          position: position,
+        );
+
+  final LightSource source;
+
+  late final Light _light = Light(
+    transform: Transform3D(),
```

Comment:
Shouldn't this be the component's transform?

Reply:
it's being set to that value the line below:
```
  Light get light {
    return _light..transform.setFrom(transform);
  }
```
makes sure they are in sync

Reply:
Wouldn't passing it be better? `Light` does not copy it and that way we don't have to use more memory.

Reply:
I see. I was concerned with someone modifying the reference or something. but I guess it is final on the Component so should work

Reply:
They still can :P but that is fine as that is the same as modifying the transform of the component.

Reply:
I do think we should still change this to use the transform by reference right?

Reply:
yes, I can do that!